internal/repositories/dcn.go

```diff
@@ -14,14 +17,44 @@ func DCNToAPI(d *models.DCN) *gmodel.Dcn {
 		Node:      d.Node,
 		ExpiresAt: d.Expiration.Ptr(),
 		Name:      d.Name.Ptr(),
+		VehicleID: d.VehicleID.Ptr(),
 	}
 }
 
-func (r *Repository) GetDCNByNode(ctx context.Context, node []byte) (*gmodel.Dcn, error) {
-	dcn, err := models.DCNS(models.DCNWhere.Node.EQ(node)).One(ctx, r.pdb.DBS().Reader)
+func (r *Repository) getDCNByNode(ctx context.Context, dcnParams model.DCNBy) (*gmodel.Dcn, error) {
```

Comment:
I think it's a bit strange that these accept `model.DCNBy`. That seems to invite trouble.

Could we have this one accept `[]byte` and the next one accept `string`?